sync/tryrwmutex.go

```diff
@@ -0,0 +1,103 @@
+package sync
+
+import (
+	"runtime"
+	"sync"
+	"sync/atomic"
+)
+
+const (
+	readOffset  = uint64(1)
+	writeOffset = uint64(1 << 20)
+	tryOffset   = uint64(1 << 40)
+
+	tryMask = uint64(1099511627775) // equivalent to setting the first 39 bits to '1'.
```

Comment:
I don't see why that would be necessary - random numbers are random numbers. Blake2b is just an implementation detail, and really if we were pulling it out into a separate package we'd use SHA2 b/c SHA2 has hardware acceleration support on a lot of machines.

Users shouldn't need to worry about implementation details.